brave/src/main/java/brave/internal/Lists.java

```diff
@@ -36,7 +36,7 @@
     if (list.size() == 1) return Collections.singletonList(list.get(0));
     if (isImmutable(list)) return list;
 
-    return concat(list, Collections.emptyList());
+    return Collections.unmodifiableList(new ArrayList<>(list));
```

Comment:
Don't think this changed the return type, concat returns it too, and uses the fixed length list as delegate

Reply:
concat special cased one side empty and in that special casing didn't ensure the input was immutable. that was the bug.